libdrgn/stack_trace.c

```diff
@@ -1212,10 +1214,11 @@ static struct drgn_error *drgn_get_stack_trace(struct drgn_program *prog,
 	if (drgn_program_is_userspace_process(prog)) {
 		return drgn_error_create(DRGN_ERROR_NOT_IMPLEMENTED,
 					 "stack unwinding is not yet supported for live processes");
-	} else if (!(prog->flags & DRGN_PROGRAM_IS_LINUX_KERNEL)
-		   && !drgn_program_is_userspace_core(prog)) {
-		return drgn_error_create(DRGN_ERROR_NOT_IMPLEMENTED,
-					 "stack unwinding is not supported for this program");
+	} else if (!warned_unsupported &&
+		   !(prog->flags & DRGN_PROGRAM_IS_LINUX_KERNEL)
+		     && !drgn_program_is_userspace_core(prog)) {
+		drgn_log_warning(prog, "warning: stack unwinding is not supported for custom programs");
+		warned_unsupported = true;
```

Comment:
The only reason we can't get stack traces of other program types is that we can't get their initial registers. We could still allow unwinding via `struct pt_regs`. Maybe we can move the original checks to `drgn_get_initial_registers()` in the !pt_regs case instead?

Reply:
This is a _much_ better approach, thank you!